[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/jinsanity07git/livedc/blob/main/notebooks/heartrate.ipynb#scrollTo=XgCQhLhv-Av0)

In [1]:
import os; import sys; os.system("pip install livedc") if "COLAB_JUPYTER_TRANSPORT" in os.environ else sys.path.insert(0, '../src')

In [2]:
from livedc.health.workout import fat_burning_heart_rate
fat_burning_heart_rate.rdoc()

The fat-burning zone is typically 70-80% of the maximum heart rate, which is calculated as:
    $$ \text{Maximum Heart Rate} = 220 - \text{Age} $$
    The lower and upper bounds of the fat-burning heart rate range are then calculated as:
    $$ \text{Lower Bound} = \text{Maximum Heart Rate} \times 0.7 $$
    $$ \text{Upper Bound} = \text{Maximum Heart Rate} \times 0.8 $$

    Args:
        age (int): Age of the person.
        
    Returns:
        tuple: A tuple containing the lower and upper bounds of the heart rate range.
    
    Reference:
    * [Cracking the Code: Heart Rate Zones for Fat Burning](https://www.miramont.us/blog/heart-rate-to-burn-fat)
    

In [3]:
age = 30
heart_rate_range = fat_burning_heart_rate(age)
mntime = 35
maxhr = 162	
avehr = 153.95

print(f"Fat-burning heart rate range for age {age}: {heart_rate_range}")

Fat-burning heart rate range for age 30: (133.0, 152.0)


In [4]:
prompt = f"If a {age}-year-old man jogs for {mntime} minutes with an average heart rate of {avehr} bpm and a max average heart rate of {maxhr} bpm" + \
            f"with a fat-burning zone of {heart_rate_range} bpm, what likely happens?"
print (prompt)

If a 30-year-old man jogs for 35 minutes with an average heart rate of 153.95 bpm and a max average heart rate of 162 bpmwith a fat-burning zone of (133.0, 152.0) bpm, what likely happens?


LLM advice DEMO:

If a 30-year-old man jogs for 35 minutes with:

- **Average HR: ~154 bpm**
- **Max HR during session: ~162 bpm**
- **Fat-burning zone: 133.0 – 152.0 bpm**

Here's what likely happens:

------

### ⚡️ **1. Exceeds Fat-Burning Zone → Enters Cardio Zone**

- His **average HR (154 bpm)** is slightly **above the fat-burning zone**, pushing him into the **aerobic/cardio zone**.
- **More total calories burned**, but **lower percentage of fat** used compared to staying within the fat-burning zone.

> 🧠 Think: **Less fat burned proportionally**, but **still a great fat-loss workout** because of the higher overall energy burn and post-exercise effects.

------

### ❤️ **2. Cardiovascular Conditioning Improves**

- Working in this range (around 80–85% of max HR) builds:
  - **Aerobic capacity (VO₂ max)**
  - **Heart strength**
  - **Lung efficiency**

This intensity is ideal for improving **endurance**, not just burning fat.

------

### 🔥 **3. Afterburn Effect (EPOC) Kicks In**

- Higher-intensity efforts like this lead to **Excess Post-Exercise Oxygen Consumption** (EPOC).
- You’ll continue burning calories **after** the workout while your body recovers.

------

### 💪 **4. Muscle Glycogen Used More Than Fat**

- At 154–162 bpm, your body shifts to **carbohydrates (glycogen)** as the main fuel source.
- Fat still contributes, but less than it would in the 133–152 bpm range.

------

### ✅ Summary:



| Metric               | Outcome                                          |
| -------------------- | ------------------------------------------------ |
| **Calories burned**  | High ✅                                           |
| **Fat % burned**     | Moderate ❗️(slightly less than in fat-burn zone)  |
| **Cardio benefit**   | Strong ✅                                         |
| **Endurance effect** | Improving ✅                                      |
| **Risk**             | Low if well-trained, moderate if unconditioned ❗️ |

------

If this session felt sustainable and left him energized (not drained), it’s an **excellent cardio endurance builder**. But for **pure fat-burning**, slightly dialing it back (e.g., aiming for 145 bpm average) would increase the **fat-burning ratio**.